x/mongo/driver/topology/topology_options.go

```diff
@@ -187,6 +187,7 @@ func WithConnString(fn func(connstring.ConnString) connstring.ConnString) Option
 					AppName:       cs.AppName,
 					Authenticator: authenticator,
 					Compressors:   cs.Compressors,
+					ServerAPI:     driver.NewServerAPIOptions().SetServerAPIVersion(driver.LatestServerAPIVersion), // FIX HANDSHAKES 1705
```

Comment:
To check, is this necessary? I see examples where the `Topology` is configured with `WithConnString` but not `WithServerOptions`.  But if we are only running the necessary integration tests, maybe this is not needed.

Reply:
I'm not sure I have enough context on `Topology`s to answer that, but I do see that `WithServerOptions` is always used alongside `WithConnString` within the "mongo", "mongo/integration", and "mongo/integration/unified" package. I suppose as long as the configured server options include `ServerAPIOptions`, then this might not be necessary (and we avoid the "default" created here).

Reply:
The main consumers of `WithConnString` that I'm aware of are internal products, so I think it's fine to not add a default here and see what the needs of those products are with regards to versioned API in the future. It should be sufficient for them to use `WithServerOptions` and set it that way.

Reply:
Sounds good! Removed the default here and on the non-auth handshaker below.